src/Integration.Vsix.UnitTests/SonarLintTagger/TaggerProviderTests.cs

```diff
@@ -117,6 +120,57 @@ public void SetUp()
                 mockSonarLanguageRecognizer.Object, mockAnalysisRequester.Object, mockTaggableBufferIndicator.Object, issueConsumerFactory.Object, logger, mockAnalysisScheduler.Object, new NoOpThreadHandler());
         }
 
+
+        #region MEF tests
+
+        [TestMethod]
+        public void CheckIsSingletonMefComponent()
+            => MefTestHelpers.CheckIsSingletonMefComponent<TaggerProvider>();
+
+        [TestMethod]
+        public void MefCtor_CheckIsExported_TaggerProvider()
+            => MefTestHelpers.CheckTypeCanBeImported<TaggerProvider, ITaggerProvider>(GetRequiredExports());
+
+        [TestMethod]
+        public void MefCtor_CheckIsExported_DocumentEvents()
+            => MefTestHelpers.CheckTypeCanBeImported<TaggerProvider, IDocumentEvents>(GetRequiredExports());
+
+        [TestMethod]
+        public void MefCtor_Check_SameInstanceExported()
+        {
+            // Importing via either exported interface should return the same instance
+            var taggerProviderImporter = new SingleObjectImporter<ITaggerProvider>();
+            var docEventsImporter = new SingleObjectImporter<IDocumentEvents>();
+            var importers = new object[] { taggerProviderImporter, docEventsImporter };
+
+            var requiredExports = GetRequiredExports();
+            var typeToExport = new Type[] { typeof(TaggerProvider) };
+
+            MefTestHelpers.Compose(importers, typeToExport, requiredExports);
+
+            taggerProviderImporter.AssertImportIsNotNull();
+            docEventsImporter.AssertImportIsNotNull();
+
+            taggerProviderImporter.Import.Should().BeSameAs(docEventsImporter.Import);
+        }
+
+        private static Export[] GetRequiredExports() => new[]
+        {
+            MefTestHelpers.CreateExport<ISonarErrorListDataSource>(),
+            MefTestHelpers.CreateExport<ITextDocumentFactoryService>(),
+            MefTestHelpers.CreateExport<IAnalyzerController>(),
+            MefTestHelpers.CreateExport<SVsServiceProvider>(),
+            MefTestHelpers.CreateExport<ISonarLanguageRecognizer>(),
+            MefTestHelpers.CreateExport<IAnalysisRequester>(),
+            MefTestHelpers.CreateExport<ITaggableBufferIndicator>(),
+            MefTestHelpers.CreateExport<IIssueConsumerFactory>(),
+            MefTestHelpers.CreateExport<ILogger>(),
+            MefTestHelpers.CreateExport<IScheduler>(),
+            MefTestHelpers.CreateExport<IThreadHandling>()
```

Comment:
This class is problematic in that it has too many dependencies. However, fixing that is a separate issue. We're not making it any worse here.